psi4/src/psi4/dlpno/mp2.cc

```diff
@@ -716,7 +725,7 @@ void DLPNOMP2::compute_df_ints() {
             }  // N loop
         }      // M loop
 
-        auto C_pao_slice = submatrix_rows(*C_pao_, riatom_to_bfs2_[centerQ]);  // TODO: PAO slices
+        auto C_pao_slice = submatrix_rows_and_cols(*C_pao_, riatom_to_bfs2_[centerQ], riatom_to_paos_ext_[centerQ]);
```

Comment:
Let me check my understanding here: `C_pao_slice` used to have dimensions `nbf_{extended domain}` by `npao(==nbf)`. Following the advice of an old comment of mine, you've now made this matrix `nbf_{extended domain}` by `npao_{extended domain}`, where `npao_{extended domain} <= npao`.

If this is the case, below in line 743 (where `C_pao_slice` is contracted with AO integrals to get PAO integrals), wouldn't you have a dimension mismatch? Now that `C_pao_slice` is smaller, I'd think you'd need a corresponding smaller slice of AO integrals. It's been a while since I looked at this code, so I could be misinterpreting things.

Also, to what degree has this change been tested? Often when you're working with these sparsity relationships, problems aren't evident until you run the code on large systems with lots of sparsity. Have you done such a test for these changes?

Reply:
Hi Zach, thank you for the review!
Concerning the dimension mismatch, this would not happen, since the index with `npao_{extended domain}` is not contracted over in the matrix multiplication in line 743.

Also, this change has been tested in my pilot DLPNO-CCSD code (PR in the near future), where 99.9% of the CCSD correlation energy is still recovered (Tested on water clusters)

Reply:
I see, it's the other dimension of `C_pao_slice` that gets contracted. Water clusters sound like a good test--lots of sparsity there